### ISSUE TITLE
Handling Timeouts

### ISSUE THREAD

**Issue by danrlee** [softonic/axios-retry#21]
Hi,

I've noticed two things that are potentially issues when handling timeouts.

You honour the original timeout across retries but the code doesn't take into account the initial request time.  If the first request took 95ms and there was a timeout of 100ms, the code would start making retry requests assuming you had the full timeout duration available, effectively doubling the timeout.

Calls are made to axios even if the timeout has dropped below 0.  This causes axios to not apply a timeout to the last request and so it waits indefinitely.

Both issues can be seen by updating the following test with new delay values and observing the timeout being passed to axios (depending on machine speed it might timeout from the request):
```js
// index.spec.js
it('should honor the original `timeout` across retries', done => {
  const client = axios.create();

  setupResponses(client, [
    () => nock('http://example.com').get('/test').delay(95).replyWithError(NETWORK_ERROR),
    () => nock('http://example.com').get('/test').delay(99).replyWithError(NETWORK_ERROR),
    () => nock('http://example.com').get('/test').delay(99).replyWithError(NETWORK_ERROR),
    () => nock('http://example.com').get('/test').reply(200)
  ]);

  axiosRetry(client, { retries: 3 });

  client.get('http://example.com/test', { timeout: 100 }).then(done.fail, error => {
    expect(error.code).toBe('ECONNABORTED');
    done();
  });
});
```
You will see two calls going to axios with the first timeout matching the one passed to axios (100), the second timeout is negative.

**Comment by rubennorte** [comment]
I see 2 issues here:

- Passing timeouts <= 0 in browsers, which means no timeout.
- Not taking the initial request into account for the timeout across retries.

I'm about to open a PR to fix both of them.

**Comment by rubennorte** [comment]
Fix released in 3.0.1
